sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java

```diff
@@ -6108,6 +6108,102 @@ public void testCountStarWithTimeInIntervalFilterLosAngeles()
     );
   }
 
+  @Test
+  public void testGroupByNullType()
+  {
+    // Cannot vectorize due to null constant expression.
+    cannotVectorize();
+    testQuery(
+        "SELECT NULL as nullcol, COUNT(*) FROM druid.foo GROUP BY 1",
+        ImmutableList.of(
+            new GroupByQuery.Builder()
+                .setDataSource(CalciteTests.DATASOURCE1)
+                .setInterval(querySegmentSpec(Filtration.eternity()))
+                .setGranularity(Granularities.ALL)
+                .setVirtualColumns(expressionVirtualColumn("v0", "null", ColumnType.STRING))
+                .setDimensions(dimensions(new DefaultDimensionSpec("v0", "d0", ColumnType.STRING)))
+                .setAggregatorSpecs(aggregators(new CountAggregatorFactory("a0")))
+                .setContext(QUERY_CONTEXT_DEFAULT)
+                .build()
+        ),
+        ImmutableList.of(
+            new Object[]{null, 6L}
+        )
+    );
+  }
+
+  @Test
+  public void testOrderByNullType()
```

Comment:
instead of overriding+disabling the testcase; it can be marked as unsupported with:
```suggestion
  @NotYetSupported(Modes.NOT_ENOUGH_RULES)
  @Test
  public void testOrderByNullType()
```
this has the benefit that it will be verified that it still fails and could also be located based on the type of failure it have

Reply:
OK, thanks, cool idea!

It is supported for the basic planner, so I kept the override in `DecoupledPlanningCalciteQueryTest` and changed the annotation from `@Disabled` to `@NotYetSupported`.